modules/indexing/src/test/java/org/apache/ignite/util/GridCommandHandlerIndexingUtils.java

```diff
@@ -69,65 +117,196 @@ private GridCommandHandlerIndexingUtils() {
 
-        try (IgniteDataStreamer<Integer, Person> streamer = ignite.dataStreamer(cacheName)) {
-            for (int i = 0; i < 10_000; i++)
-                streamer.addData(i, new Person(rand.nextInt(), valueOf(rand.nextLong())));
+        try (IgniteDataStreamer<Integer, Object> streamer = ignite.dataStreamer(cacheName)) {
+            int entity = 0;
+            for (Function<Random, Object> fun : qryEntities.values()) {
+                for (int i = 0; i < cnt; i++)
+                    streamer.addData(i + (entity * cnt), fun.apply(rand));
+
+                streamer.flush();
+                entity++;
+            }
         }
     }
 
     /**
-     * Create query entity.
+     * Deleting a rows from the cache without updating indexes.
+     *
+     * @param log Logger.
+     * @param internalCache Cache.
+     * @param partId Partition number.
+     * @param filter Entry filter.
      */
-    private static QueryEntity personEntity() {
-        QueryEntity entity = new QueryEntity();
+    static <K, V> void breakCacheDataTree(
+        IgniteLogger log,
+        IgniteInternalCache<K, V> internalCache,
+        int partId,
+        @Nullable BiPredicate<Integer, Entry<K, V>> filter
+    ) {
+        requireNonNull(log);
+        requireNonNull(internalCache);
+
+        GridCacheContext<K, V> cacheCtx = internalCache.context();
+
+        GridDhtLocalPartition dhtLocPart = cacheCtx.dht().topology().localPartition(partId);
+
+        CacheDataStore cacheDataStore = cacheCtx.group().offheap().dataStore(dhtLocPart);
+
+        String delegate = "delegate";
+        if (hasField(cacheDataStore, delegate))
+            cacheDataStore = field(cacheDataStore, delegate);
+
+        CacheDataRowStore cacheDataRowStore = field(cacheDataStore, "rowStore");
+        CacheDataTree cacheDataTree = field(cacheDataStore, "dataTree");
+
+        String cacheName = internalCache.name();
+
+        QueryCursor<Entry<K, V>> qryCursor = cacheCtx.kernalContext().grid().cache(cacheName).withKeepBinary()
+            .query(new ScanQuery<>(partId));
+
+        Iterator<Entry<K, V>> cacheEntryIter = qryCursor.iterator();
+
+        IgniteCacheDatabaseSharedManager db = cacheCtx.shared().database();
+        int cacheId = CU.cacheId(cacheName);
+        int i = 0;
+
+        while (cacheEntryIter.hasNext()) {
+            Entry<K, V> entry = cacheEntryIter.next();
+
+            if (nonNull(filter) && !filter.test(i++, entry))
+                continue;
+
+            db.checkpointReadLock();
+
+            try {
+                CacheDataRow oldRow = cacheDataTree.remove(
+                    new SearchRow(cacheId, cacheCtx.toCacheKeyObject(entry.getKey()))
+                );
+
+                if (nonNull(oldRow))
+                    cacheDataRowStore.removeRow(oldRow.link(), INSTANCE);
+            }
+            catch (IgniteCheckedException e) {
+                log.error("Failed to remove key skipping indexes: " + entry, e);
```

Comment:
Maybe it's not your code but I still wonder - why we ignore this exception in test? Shouldn't we just fail current test instead?

Reply:
You're right, I changed it to an exception.